src/go/wsl-helper/cmd/kubeconfig.go

```diff
@@ -47,68 +45,91 @@ var kubeconfigCmd = &cobra.Command{
-					return fmt.Errorf("failed to remove kubeconfig symlink: %w", err)
-				}
-			}
-		}
-
-		if !enable {
-			return nil
+		_, err := os.Stat(configPath)
+		if err != nil {
+			return fmt.Errorf("could not open Windows kubeconfig: %w", err)
 		}
-
 		cmd.SilenceUsage = true
 
-		winConfig, err := readKubeConfig(winConfigPath)
-		if err != nil {
-			return err
+		configDir := path.Join(homedir.HomeDir(), ".kube")
+		linkPath := path.Join(configDir, "config")
+		manualSymlink, symlinkErr := requireManualSymlink(linkPath)
+		if !manualSymlink && symlinkErr != nil {
+			return symlinkErr
 		}
 
-		linuxConfig, err := readKubeConfig(linuxConfigPath)
-		if err != nil && !errors.Is(err, os.ErrNotExist) {
-			return err
+		if enable {
+			err = os.Mkdir(configDir, 0o750)
+			if err != nil && !errors.Is(err, os.ErrExist) {
+				// The error already contains the full path, we can't do better.
+				return err
+			}
+			err = os.Symlink(configPath, linkPath)
+			if err != nil {
+				if errors.Is(err, os.ErrExist) {
+					// If it already exists, do nothing; even if it's not a symlink.
+					return nil
+				}
+				return err
+			}
+		} else {
+			// Config contains user configuration
+			if manualSymlink {
+				return symlinkErr
+			}
+			// No need to create if we want to remove it
+			target, err := os.Readlink(linkPath)
+			if err != nil {
+				if errors.Is(err, os.ErrNotExist) {
+					return nil
+				}
+				return err
+			}
+			if target == configPath {
+				if err = removeConfig(linkPath); err != nil {
+					return err
+				}
+			}
 		}
+		return nil
+	},
+}
 
-		cleanConfig := removeExistingRDConfig(rdCluster, &linuxConfig)
-
-		kubeConfig, err := updateKubeConfig(winConfig, *cleanConfig, rdNetworking)
-		if err != nil {
-			return fmt.Errorf("failed to construct kubeconfig: %w", err)
+// requireManualSymlink checks the config to determine if it contains a single entry for Contexts, Clusters, and Users.
+// If all three are named 'rancher-desktop', we assume that this configuration was written by Rancher Desktop 1.12,
+// and we can remove it and replace it with a symlink. If a user's config contains Rancher Desktop's specific configuration
+// along with user-provided config, or if it only contains user-provided config, we return a boolean and an error.
+// This indicates through diagnostics to the user that manual action is required.
+func requireManualSymlink(linkPath string) (bool, error) {
+	// Check to see if config is rancher desktop only
+	if exitingCfg, err := readKubeConfig(linkPath); err == nil {
+		if len(exitingCfg.Contexts) == 1 && exitingCfg.Contexts[0].Name == rdCluster &&
+			len(exitingCfg.Clusters) == 1 && exitingCfg.Clusters[0].Name == rdCluster &&
+			len(exitingCfg.Users) == 1 && exitingCfg.Users[0].Name == rdCluster {
+			if err := removeConfig(linkPath); err != nil {
+				return false, err
+			}
+		} else {
+			return true, fmt.Errorf("could not create a symlink for the following kubeconfig file %s: %+v", linkPath, exitingCfg)
```

Comment:
The `%+v` is the existing kubeconfig; that might be too large to show as an error message?